relay-sampling/src/lib.rs

```diff
@@ -373,32 +377,11 @@ impl SamplingRule {
         if !self.is_decaying_rule() {
             return true;
         }
-        if !self.is_idle() && !self.is_expired() {
-            return true;
+        if let Some(time_range) = self.time_range.as_ref() {
```

Comment:
This check is still redundant with `self.is_decaying_rule()`. Is `is_decaying_rule` used anywhere else? If not, I would remove it and just do a `match` statement here.